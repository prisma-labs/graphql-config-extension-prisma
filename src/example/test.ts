import { patchConfig } from '../index'

test('patches endpoints and directives (to all projects if one of them has prisma) to config', async () => {
  expect.assertions(1);
  const config = {
    config: {
      projects: {
        app: {
          extensions: {}
        },
        database: {
          extensions: {
            prisma: 'prisma.yml',
            graphcool: 'graphcool.yml',
          },
        },
      },
    }
  } as any
  const newConfig = await patchConfig(config, `${__dirname}`);
  expect(newConfig).toMatchSnapshot();
});