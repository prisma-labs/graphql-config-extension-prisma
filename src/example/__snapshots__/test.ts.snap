// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patches endpoints and directives (to all projects if one of them has prisma) to config 1`] = `
Object {
  "config": Object {
    "projects": Object {
      "app": Object {
        "extensions": Object {
          "customDirectives": Array [
            "enum DeleteValues { CASCADE SET_NULL }",
            "directive @default(value: String!) on FIELD_DEFINITION | SCALAR",
            "directive @relation(name: String!, onDelete: DeleteValues) on FIELD_DEFINITION | SCALAR",
            "directive @unique on FIELD_DEFINITION | SCALAR",
            "directive @pgRelation(column: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgRelationTable(table: String!, relationColumn: String!, targetColumn: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgTable(name: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgColumn(name: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgDefault(value: String!) on FIELD_DEFINITION | SCALAR",
          ],
        },
      },
      "database": Object {
        "extensions": Object {
          "customDirectives": Array [
            "enum DeleteValues { CASCADE SET_NULL }",
            "directive @default(value: String!) on FIELD_DEFINITION | SCALAR",
            "directive @relation(name: String!, onDelete: DeleteValues) on FIELD_DEFINITION | SCALAR",
            "directive @unique on FIELD_DEFINITION | SCALAR",
            "directive @pgRelation(column: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgRelationTable(table: String!, relationColumn: String!, targetColumn: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgTable(name: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgColumn(name: String!) on FIELD_DEFINITION | SCALAR",
            "directive @pgDefault(value: String!) on FIELD_DEFINITION | SCALAR",
          ],
          "endpoints": Object {
            "dev": Object {
              "headers": undefined,
              "url": "https://eu1.prisma.sh/public-asd/n/dev",
            },
          },
          "graphcool": "graphcool.yml",
          "prisma": "prisma.yml",
        },
      },
    },
  },
}
`;
